test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+    {
+        uresource.name = resource_value.to_owned();
+    } else {
+        error!("Error: name field is not a string in resource");
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("instance"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.instance = Some(resource_value.to_owned());
+    } else {
+        error!("Error: instance field is not a string in resource");
+        //Some("None".to_owned())
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("message"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.message = Some(resource_value.to_owned());
+    } else {
+        error!("Error: message field is not a string in resource_value");
+        //Some("None".to_owned())
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("id"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        if let Ok(parsed_id) = resource_value.parse::<u32>() {
+            uresource.id = Some(parsed_id);
+        } else {
+            error!("Error: id field parsing to u32");
+            // Some(0)
+        }
+    } else {
+        error!("Error: id field is not string");
+        //Some(0)
+    };
+    uresource
+}
+
+fn parse_string_field(value: &Value, field: &str) -> Result<String, serde_json::Error> {
+    if let Some(entity_value) = value
+        .get("entity")
+        .and_then(|entity_value| entity_value.get(field))
+        .and_then(|field_value| field_value.as_str())
+    {
+        Ok(entity_value.to_owned())
+    } else {
+        error!("Error: {field} field is not a string");
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_u32_field(value: &Value, field: &str) -> Result<Option<u32>, serde_json::Error> {
+    if let Some(entity_value) = value
+        .get("entity")
+        .and_then(|entity_value| entity_value.get(field))
+        .and_then(|field_value| field_value.as_str())
+    {
+        if let Ok(parsed_value) = entity_value.parse::<u32>() {
+            Ok(Some(parsed_value))
+        } else {
+            error!("Error: {field} is not a number");
+            Err(serde::de::Error::custom(format!(
+                "Error: {field} is not a number"
+            )))
+        }
+    } else {
+        error!("Error: {} field is not a string", field);
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_uentity(value: &Value) -> UEntity {
+    let name = match parse_string_field(value, "name") {
+        Ok(value) => value,
+        Err(_) => "None".to_owned(),
+    };
+
+    let id = match parse_u32_field(value, "id") {
```

Comment:
Consider this instead, since the error is not used or propagated:
```rust
let id = parse_u32_field(value, "id").map_or(None, |v| Some(v));
```

Reply:
did something to match this, couldn't get your exact code to work, so changed it to some sort of .ok().flatten()... not sure if that's ideal, let's take a look.